code/datums/mutations/holy_mutation/honorbound.dm

```diff
@@ -0,0 +1,125 @@
+	//* is not a gun and out of range of attacking
+	if(!honorbound.DirectAccess(clickingon) && !isgun(weapon))
+		return
+	if(!honorbound.combat_mode && ((!weapon || !weapon.force) && !LAZYACCESS(modifiers, RIGHT_CLICK)))
+		return
+	var/mob/living/clickedmob = clickingon
+	if(!is_honorable(honorbound, clickedmob))
+		return COMSIG_MOB_CANCEL_CLICKON
+
+/datum/mutation/human/honorbound/proc/guilty(mob/living/user)
+	if(user in guilty)
+		return
+	var/datum/mind/guilty_conscience = user.mind
+	if(guilty_conscience) //sec and medical are immune to becoming guilty through attack (we don't check holy because holy shouldn't be able to attack eachother anyways)
+		var/job = guilty_conscience.assigned_role
+		if(job in (GLOB.security_positions + GLOB.medical_positions))
+			return
+	to_chat(owner, "<span class='notice'>[user] is now considered guilty by [GLOB.deity] for attacking you first.</span>")
+	guilty += user
+
+/datum/mutation/human/honorbound/proc/is_honorable(mob/living/carbon/human/honorbound_human, mob/living/target_creature)
+	var/is_guilty = (target_creature in guilty)
+	//THE UNREADY (Applies over ANYTHING else!)
+	if(target_creature.IsSleeping() || HAS_TRAIT(target_creature, TRAIT_RESTRAINED))
+		to_chat(honorbound_human, "<span class='warning'>There is no honor in attacking the <b>unready</b>.</span>")
+		return FALSE
+	//THE JUST (Applies over guilt except for med, so you best be careful!)
+	if(ishuman(target_creature))
+		var/mob/living/carbon/human/target_human = target_creature
+		var/job = target_human.mind?.assigned_role
+		var/is_holy = target_human.mind?.holy_role
+		if(job in GLOB.security_positions || is_holy)
+			to_chat(honorbound_human, "<span class='warning'>There is nothing rightous in attacking the <b>just</b>.</span>")
+			return FALSE
+		if(job in GLOB.medical_positions)
+			to_chat(honorbound_human, "<span class='warning'>If you truly think this healer is not <b>innocent</b>, declare them guilty.</span>")
+			return FALSE
+	//THE INNOCENT
+	if(!is_guilty)
+		to_chat(honorbound_human, "<span class='warning'>There is nothing rightous in attacking the <b>innocent</b>.</span>")
+		return FALSE
+
+// SIGNALS THAT ARE FOR BEING ATTACKED FIRST (GUILTY)
+/datum/mutation/human/honorbound/proc/attackby_guilt(datum/source, obj/item/I, mob/attacker)
+	SIGNAL_HANDLER
+	if(I.force && I.damtype != STAMINA)
+		guilty(attacker)
+
+/datum/mutation/human/honorbound/proc/hulk_guilt(datum/source, mob/attacker)
+	SIGNAL_HANDLER
+	guilty(attacker)
+
+/datum/mutation/human/honorbound/proc/hand_guilt(datum/source, mob/living/attacker)
+	SIGNAL_HANDLER
+	if(attacker.combat_mode)
+		guilty(attacker)
+
+/datum/mutation/human/honorbound/proc/paw_guilt(datum/source, mob/living/attacker)
+	SIGNAL_HANDLER
+	guilty(attacker)
+
+/datum/mutation/human/honorbound/proc/bullet_guilt(datum/source, obj/projectile/Proj)
+	SIGNAL_HANDLER
+	var/mob/living/shot_honorbound = source
+	if(istype(Proj, /obj/projectile/beam)||istype(Proj, /obj/projectile/bullet))
```

Comment:
does this really need to only count beams and bullets? what about magical projectiles and the like?